posts/2021-03-24-microprofile40-open-liberty-21003.adoc

```diff
@@ -0,0 +1,442 @@
+
+MicroProfile Health allows services to report their health, and publish overall health status to a defined endpoint. A service reports "UP" if it is available and reports "DOWN" if it is unavailable. In addition to reporting the state of individual services, MicroProfile Health also aggregates the overall status as "UP" if all the services are "UP". If one of the services is "DOWN" then the overall status is also "DOWN". A service orchestrator can then use the health statuses to make decisions.
+
+MicroProfile Health checks its own health by performing necessary self-checks and then reports its overall status by implementing the API provided by MicroProfile Health. A self-check can be a check on anything that the service needs, such as a dependency, a successful connection to an endpoint, a system property, a database connection, or the availability of required resources. MicroProfile offers checks for both liveness and readiness.
+
+Significant changes in MicroProfile Health 3.0:
+
+* The overall default Readiness status was changed to "DOWN", with an empty response until all the deployed application(s) have been started. A new MicroProfile Config property (`mp.health.default.readiness.empty.response=UP`) is introduced to change the overall default Readiness check status to "UP", during application start up, that do not have any user-defined health checks.
+
+* The `HealthCheckResponseBuilder.state(Boolean UP)` method was also renamed to `HealthCheckResponseBuilder.status(Boolean UP)` for HealthCheckResponse deserialization compatibility, where the JSON health check response string can now be deserialized into an HealthCheckResponse object.
+
+* The deprecated `@Health` qualifier was removed. Instead, the `@Liveness` or `@Readiness` qualifiers should be used as appropriate in your HealthCheck implementations.
+
+Applications are expected to provide health check procedures by implementing the HealthCheck interface with the `@Liveness` or `@Readiness` annotations. These are used by Open Liberty to verify the Liveness or Readiness of the application, respectively. Add the logic of your health check in the call() method, and return an instance of `HealthCheckResponse` by using the simple `up` and `down` API methods:
+
+[source, java]
+----
+//**Liveness Check**
+@Liveness
+@ApplicationScoped
+public class AppLiveCheck implements HealthCheck {
+...
+    @Override
+     public HealthCheckResponse call() {
+       ...
+       HealthCheckResponse.up("my-liveness-check");
+       ...
+     }
+}
+
+//**Readiness Check**
+@Readiness
+@ApplicationScoped
+public class AppReadyCheck implements HealthCheck {
+...
+    @Override
+     public HealthCheckResponse call() {
+       ...
+       HealthCheckResponse.named("my-app-readiness").status(isMyAppReady()).build();
+       ...
+     }
+}
+...
+----
+
+To view the status of each health check, access the either the `http://<hostname>:<port>/health/live` or `http://<hostname>:<port>/health/ready endpoints`.
```

Comment:
This sentence does not make sense - "access the either the"?

Reply:
```suggestion
To view the status of each health check, access either `http://<hostname>:<port>/health/live` or `http://<hostname>:<port>/health/ready endpoints`.
```